test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -598,6 +605,30 @@ private static String groupName(ThreadGroup threadGroup) {
         return tempList.subList(0, size);
     }
 
+    public static XContentBuilder shuffleXContent(XContentBuilder builder, Set<String> exceptFieldNames) throws IOException {
+        BytesReference bytes = builder.bytes();
+        XContentParser parser = XContentFactory.xContent(bytes).createParser(bytes);
+        // use ordered maps for reproducibility
+        Map<String, Object> shuffledMap = shuffleMap(parser.mapOrdered(), exceptFieldNames, random());
+        XContentBuilder jsonBuilder = XContentFactory.jsonBuilder();
+        return jsonBuilder.map(shuffledMap);
+    }
+
+    private static Map<String, Object> shuffleMap(Map<String, Object> map, Set<String> exceptFieldNames, Random r) {
+        List<String> keys = new ArrayList<>(map.keySet());
+        Map<String, Object> targetMap = new TreeMap<>();
```

Comment:
I think you have to sort before shuffling, otherwise it will not be deterministic, cause the order in the list still depends on the order in the set, which depends on the jvm :)